### PR TITLE
fix two benchmarks

### PR DESCRIPTION
**Pull request by naivewong**
- BenchmarkCompaction will always OOM even in Prometheus test machine
- BenchmarkCreateSeries has a high probability to fail

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
most likely because of the `genSeries` , can you check if we can optimize it or if not will merge this change.

File: head_test.go [hunk 1]
```diff
@@ -34,7 +34,11 @@ import (
 )
 
 func BenchmarkCreateSeries(b *testing.B) {
-	lbls, err := labels.ReadLabels(filepath.Join("testdata", "20kseries.json"), b.N)
+	n := b.N
+	if n > 20000 {
+		n = 20000
+	}
+	lbls, err := labels.ReadLabels(filepath.Join("testdata", "20kseries.json"), n)
```
Comment (krasi-georgiev) [inline head_test.go]: can we use `genSeries` here?

**Comment by codesome** [comment]
@krasi-georgiev yes, while benchmarking compaction recently myself I found that creation of blocks takes a lot of memory. I plan to optimize it in the future.

File: head_test.go (hunk 1 above)
Comment (codesome) [reply head_test.go]: We may need some options for `genSeries` in that case. For example `20kseries.json` has changing label names for series (label names do repeat), but `genSeries` will have the same label names for all the series.

**Comment by krasi-georgiev** [comment]
@codesome is it ok to merge this change or shall @naivewong try to optimize genSeries instead?

**Comment by codesome** [comment]
@krasi-georgiev It can be merged now. We can visit that later.

**Pull request merged**